File: 2856-count-complete-subarrays-in-an-array/2856-count-complete-subarrays-in-an-array.py
class Solution:
    def countCompleteSubarrays(self, nums: List[int]) -> int:
        d=len(set(nums))
        n=len(nums)

        freq=defaultdict(int)
        i=j=ans=0

        while(i<n):
            freq[nums[i]] +=1

            while len(freq)==d:
                ans +=(n-i)

                freq[nums[j]] -=1
                if(freq[nums[j]]==0):
                    del freq[nums[j]]
                j+=1
            i +=1
        return ans


        

File: 2856-count-complete-subarrays-in-an-array/2856-count-complete-subarrays-in-an-array.java
class Solution {
    public int countCompleteSubarrays(int[] nums) {
        int n = nums.length;
        int ans = 0, d = 0;

        boolean[] exist = new boolean[2001];
        for (var x : nums) {
            if (!exist[x]) {
                exist[x] = true;
                d++;
            }
        }
        HashSet<Integer> ch = new HashSet<>();

        for (int i = 0; i < n; i++) {
            ch.clear();

            for (int j = i; j < n; j++) {
                ch.add(nums[j]);
                if (ch.size() == d)
                    ans++;
            }
        }
        return ans;
    }
}